concrete/src/Routing/RoutingServiceProvider.php

```diff
@@ -10,7 +10,7 @@ class RoutingServiceProvider extends Provider
      */
     public function register()
     {
-        $this->app->singleton(RouterInterface::class);
+        $this->app->singleton(Router::class);
         $this->app->bind(RouterInterface::class, Router::class);
```

Comment:
Here's the test I put in my app.php:

```php
<?php

use Concrete\Core\Routing\Router;
use Concrete\Core\Routing\RouterInterface;

$app->singleton(Router::class);
$app->alias(Router::class, RouterInterface::class);


class CustomRouter extends Router {}

$app->alias(CustomRouter::class, RouterInterface::class);

$instance1 = $app->make(RouterInterface::class);
$instance2 = $app->make(RouterInterface::class);
$instance3 = $app->make(CustomRouter::class);
$instance4 = $app->make(CustomRouter::class);

dd(($instance1 === $instance2 && $instance2 === $instance3 && $instance3 === $instance4));
```

That outputs false because your singleton is tied to `Router::class`. Once you rebind RouterInterface there is no singleton in the mix so you end up with 4 different instances.